src/pagination/Pagination.ts

```diff
@@ -0,0 +1,175 @@
+  public get renderBullet() { return this._renderBullet; }
+  public get renderFraction() { return this._renderFraction; }
+  public get fractionCurrentFormat() { return this._fractionCurrentFormat; }
+  public get fractionTotalFormat() { return this._fractionTotalFormat; }
+  public get scrollOnChange() { return this._scrollOnChange; }
+
+  public set parentEl(val: PaginationOptions["parentEl"]) { this._parentEl = val; }
+  public set selector(val: PaginationOptions["selector"]) { this._selector = val; }
+  public set type(val: PaginationOptions["type"]) { this._type = val; }
+  public set bulletWrapperclassPrefixClass(val: PaginationOptions["classPrefix"]) { this._classPrefix = val; }
+  public set dynamicBulletCount(val: PaginationOptions["dynamicBulletCount"]) { this._dynamicBulletCount = val; }
+  public set dynamicFlickingOptions(val: PaginationOptions["dynamicFlickingOptions"]) { this._dynamicFlickingOptions = val; }
+  public set renderBullet(val: PaginationOptions["renderBullet"]) { this._renderBullet = val; }
+  public set renderFraction(val: PaginationOptions["renderFraction"]) { this._renderFraction = val; }
+  public set fractionCurrentFormat(val: PaginationOptions["fractionCurrentFormat"]) { this._fractionCurrentFormat = val; }
+  public set fractionTotalFormat(val: PaginationOptions["fractionTotalFormat"]) { this._fractionTotalFormat = val; }
+  public set scrollOnChange(val: PaginationOptions["scrollOnChange"]) { this._scrollOnChange = val; }
+
+  public constructor({
+    parentEl = null,
+    selector = PAGINATION.SELECTOR,
+    type = PAGINATION.TYPE.BULLET,
+    classPrefix = PAGINATION.PREFIX,
+    dynamicBulletCount = 5,
+    dynamicFlickingOptions = {},
+    renderBullet = (className: string) => `<span class="${className}"></span>`,
+    renderFraction = (currentClass: string, totalClass: string) => `<span class="${currentClass}"></span>/<span class="${totalClass}"></span>`,
+    fractionCurrentFormat = (index: number) => index.toString(),
+    fractionTotalFormat = (index: number) => index.toString(),
+    scrollOnChange = (index: number, ctx: ScrollContext) => ctx.moveTo(index)
+  }: Partial<PaginationOptions> = {}) {
+    this._parentEl = parentEl;
+    this._selector = selector;
+    this._type = type;
+    this._classPrefix = classPrefix;
+    this._dynamicBulletCount = dynamicBulletCount;
+    this._dynamicFlickingOptions = dynamicFlickingOptions;
+    this._renderBullet = renderBullet;
+    this._renderFraction = renderFraction;
+    this._fractionCurrentFormat = fractionCurrentFormat;
+    this._fractionTotalFormat = fractionTotalFormat;
+    this._scrollOnChange = scrollOnChange;
+  }
+
+  public init(flicking: Flicking): void {
+    if (this._flicking) {
+      this.destroy();
+    }
+
+    this._flicking = flicking;
+
+    const type = this._type;
+    const selector = this._selector;
+    const parentEl = this._parentEl ? this._parentEl : flicking.element;
+    const wrapper = parentEl.querySelector(selector);
```

Comment:
Isn't there a case of directly using an element other than a selector?

Reply:
That's not supported.
It automatically detects element within the viewport element, also can search element from other element with the `parentEl` option
I thought it would be easier to use especially in the frameworks.